### PR TITLE
MM-14575: Automatically serve static files for plugins

### PR DESCRIPTION
**Pull request by happygaijin**
* Added sample public file
* Added HAS_PUBLIC to Makefile to test for public folder
* Added "static_files" config setting
* Related to:
 [https://github.com/mattermost/mattermost-server/pull/10476](https://github.com/mattermost/mattermost-server/pull/10476l)
[https://github.com/mattermost/mattermost-server/issues/10447](https://github.com/mattermost/mattermost-server/issues/10447)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,6 +75,10 @@ ifneq ($(HAS_SERVER),)
 	cd server && env GOOS=linux GOARCH=amd64 $(GO) build -o dist/plugin-linux-amd64;
 	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build -o dist/plugin-darwin-amd64;
 	cd server && env GOOS=windows GOARCH=amd64 $(GO) build -o dist/plugin-windows-amd64.exe;
+ifneq ($(HAS_PUBLIC),)
+	mkdir -p server/dist/public;
```
Comment (hanzei) [inline Makefile]: What do you think about pre creating this, similar to `assets` https://github.com/mattermost/mattermost-plugin-sample/tree/master/assets?

File: Makefile [hunk 2]
```diff
@@ -75,6 +75,10 @@ ifneq ($(HAS_SERVER),)
...
+	cp -r server/public/* server/dist/public/;
```
Comment (hanzei) [inline Makefile]: This should be part of `bundle`

File: build/setup.mk [hunk 3]
```diff
@@ -25,6 +25,9 @@ HAS_SERVER ?= $(shell build/bin/manifest has_server)
 # Determine if a webapp is defined in the manifest.
 HAS_WEBAPP ?= $(shell build/bin/manifest has_webapp)
 
+# Determine if a /public folder is in use
+HAS_PUBLIC ?= $(wildcard server/public/*)
```
Comment (hanzei) [inline build/setup.mk]: Would you please change this to 
```suggestion
HAS_PUBLIC ?= $(wildcard public/.)
```

File: Makefile (hunk 1 above)
Comment (happygaijin) [reply Makefile]: Yeah, especially with getting rid of the static_files setting in the plugin.json. It makes sense.

File: build/setup.mk (hunk 3 above)
Comment (happygaijin) [reply build/setup.mk]: Sure. The destination on the server would then just be the root of the plugin folder?

Comment (hanzei) [reply build/setup.mk]: If you are talking about the destination in the bundle: Yes.

Comment (happygaijin) [reply build/setup.mk]: 👍

**Comment by levb** [comment]
@happygaijin @hanzei This can be merged regardless of the other PRs, right?

**Comment by hanzei** [comment]
@levb Yes

**Pull request merged**